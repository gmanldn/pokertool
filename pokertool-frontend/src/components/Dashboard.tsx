# POKERTOOL-HEADER-START
# ---
# schema: pokerheader.v1
# project: pokertool
# file: pokertool-frontend/src/components/Dashboard.tsx
# version: v28.0.0
# last_commit: '2025-09-23T08:41:38+01:00'
# fixes:
# - date: '2025-09-25'
#   summary: Enhanced enterprise documentation and comprehensive unit tests added
# ---
# POKERTOOL-HEADER-END
import React, { useEffect, useState } from 'react';
import {
  Grid,
  Paper,
  Typography,
  Box,
  Card,
  CardContent,
  IconButton,
  Chip,
  LinearProgress,
  useTheme,
  useMediaQuery,
} from '@mui/material';
import {
  TrendingUp,
  TrendingDown,
  AttachMoney,
  Casino,
  Timer,
  Refresh,
} from '@mui/icons-material';
import { Line, Doughnut } from 'react-chartjs-2';
import {
  Chart as ChartJS,
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend,
  ArcElement,
} from 'chart.js';
import { WebSocketMessage } from '../hooks/useWebSocket';

ChartJS.register(
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend,
  ArcElement
);

interface DashboardProps {
  messages: WebSocketMessage[];
}

interface SessionStats {
  handsPlayed: number;
  vpip: number;
  pfr: number;
  aggression: number;
  winRate: number;
  profit: number;
  duration: string;
}

export const Dashboard: React.FC<DashboardProps> = ({ messages }) => {
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));

  const [stats, setStats] = useState<SessionStats>({
    handsPlayed: 0,
    vpip: 0,
    pfr: 0,
    aggression: 0,
    winRate: 0,
    profit: 0,
    duration: '00:00:00',
  });

  const [loading, setLoading] = useState(false);
  const [lastUpdate, setLastUpdate] = useState(new Date());

  useEffect(() => {
    // Process real-time updates from WebSocket
    const latestStatsMessage = messages
      .filter(msg => msg.type === 'stats_update')
      .pop();
    
    if (latestStatsMessage) {
      setStats(prevStats => ({
        ...prevStats,
        ...latestStatsMessage.data,
      }));
      setLastUpdate(new Date());
    }
  }, [messages]);

  const handleRefresh = () => {
    setLoading(true);
    // Simulate refresh
    setTimeout(() => {
      setLoading(false);
      setLastUpdate(new Date());
    }, 1000);
  };

  const profitChartData = {
    labels: ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'],
    datasets: [
      {
        label: 'Profit ($)',
        data: [120, 190, 80, 250, 200, 400, 320],
        borderColor: theme.palette.primary.main,
        backgroundColor: `${theme.palette.primary.main}33`,
        tension: 0.4,
      },
    ],
  };

  const gameTypeData = {
    labels: ['Cash Games', 'Tournaments', 'Sit & Go'],
    datasets: [
      {
        data: [60, 30, 10],
        backgroundColor: [
          theme.palette.primary.main,
          theme.palette.secondary.main,
          theme.palette.warning.main,
        ],
      },
    ],
  };

  const StatCard = ({ title, value, icon, trend, color }: any) => (
    <Card
      sx={{
        height: '100%',
        background: theme.palette.mode === 'dark'
          ? 'linear-gradient(145deg, #1e1e1e 0%, #2d2d2d 100%)'
          : 'linear-gradient(145deg, #ffffff 0%, #f5f5f5 100%)',
        transition: 'transform 0.2s',
        '&:hover': {
          transform: 'translateY(-4px)',
        },
      }}
    >
      <CardContent>
        <Box display="flex" justifyContent="space-between" alignItems="center">
          <Box>
            <Typography color="textSecondary" variant="subtitle2" gutterBottom>
              {title}
            </Typography>
            <Typography variant="h4" fontWeight="bold">
              {value}
            </Typography>
            {trend !== undefined && (
              <Box display="flex" alignItems="center" mt={1}>
                {trend > 0 ? (
                  <TrendingUp color="success" fontSize="small" />
                ) : (
                  <TrendingDown color="error" fontSize="small" />
                )}
                <Typography
                  variant="body2"
                  color={trend > 0 ? 'success.main' : 'error.main'}
                  ml={0.5}
                >
                  {Math.abs(trend)}%
                </Typography>
              </Box>
            )}
          </Box>
          <Box
            sx={{
              backgroundColor: `${color}22`,
              borderRadius: 2,
              p: 1.5,
            }}
          >
            {icon}
          </Box>
        </Box>
      </CardContent>
    </Card>
  );

  return (
    <Box sx={{ flexGrow: 1, p: isMobile ? 2 : 3 }}>
      {loading && <LinearProgress />}
      
      <Box display="flex" justifyContent="space-between" alignItems="center" mb={3}>
        <Typography variant={isMobile ? 'h5' : 'h4'} fontWeight="bold">
          Dashboard
        </Typography>
        <Box display="flex" alignItems="center" gap={2}>
          <Chip
            label={`Last Update: ${lastUpdate.toLocaleTimeString()}`}
            size="small"
            variant="outlined"
          />
          <IconButton onClick={handleRefresh} size="small">
            <Refresh />
          </IconButton>
        </Box>
      </Box>

      <Grid container spacing={3}>
        {/* Stat Cards */}
        <Grid item xs={12} sm={6} md={3}>
          <StatCard
            title="Hands Played"
            value={stats.handsPlayed.toLocaleString()}
            icon={<Casino sx={{ fontSize: 28, color: theme.palette.primary.main }} />}
            trend={12}
            color={theme.palette.primary.main}
          />
        </Grid>
        
        <Grid item xs={12} sm={6} md={3}>
          <StatCard
            title="Session Profit"
            value={`$${stats.profit.toLocaleString()}`}
            icon={<AttachMoney sx={{ fontSize: 28, color: theme.palette.success.main }} />}
            trend={stats.profit > 0 ? 8 : -5}
            color={theme.palette.success.main}
          />
        </Grid>
        
        <Grid item xs={12} sm={6} md={3}>
          <StatCard
            title="Win Rate"
            value={`${stats.winRate}%`}
            icon={<TrendingUp sx={{ fontSize: 28, color: theme.palette.info.main }} />}
            trend={3}
            color={theme.palette.info.main}
          />
        </Grid>
        
        <Grid item xs={12} sm={6} md={3}>
          <StatCard
            title="Session Time"
            value={stats.duration}
            icon={<Timer sx={{ fontSize: 28, color: theme.palette.warning.main }} />}
            color={theme.palette.warning.main}
          />
        </Grid>

        {/* Profit Chart */}
        <Grid item xs={12} lg={8}>
          <Paper
            sx={{
              p: 3,
              height: isMobile ? 300 : 400,
              background: theme.palette.mode === 'dark'
                ? 'linear-gradient(145deg, #1e1e1e 0%, #2d2d2d 100%)'
                : 'white',
            }}
          >
            <Typography variant="h6" gutterBottom>
              Weekly Profit Trend
            </Typography>
            <Box sx={{ height: isMobile ? 240 : 340 }}>
              <Line
                data={profitChartData}
                options={{
                  responsive: true,
                  maintainAspectRatio: false,
                  plugins: {
                    legend: {
                      display: false,
                    },
                  },
                  scales: {
                    y: {
                      grid: {
                        color: theme.palette.divider,
                      },
                      ticks: {
                        color: theme.palette.text.secondary,
                      },
                    },
                    x: {
                      grid: {
                        display: false,
                      },
                      ticks: {
                        color: theme.palette.text.secondary,
                      },
                    },
                  },
                }}
              />
            </Box>
          </Paper>
        </Grid>

        {/* Game Type Distribution */}
        <Grid item xs={12} lg={4}>
          <Paper
            sx={{
              p: 3,
              height: isMobile ? 300 : 400,
              background: theme.palette.mode === 'dark'
                ? 'linear-gradient(145deg, #1e1e1e 0%, #2d2d2d 100%)'
                : 'white',
            }}
          >
            <Typography variant="h6" gutterBottom>
              Game Type Distribution
            </Typography>
            <Box
              sx={{
                height: isMobile ? 240 : 340,
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'center',
              }}
            >
              <Doughnut
                data={gameTypeData}
                options={{
                  responsive: true,
                  maintainAspectRatio: false,
                  plugins: {
                    legend: {
                      position: 'bottom',
                      labels: {
                        color: theme.palette.text.primary,
                      },
                    },
                  },
                }}
              />
            </Box>
          </Paper>
        </Grid>

        {/* Quick Stats */}
        <Grid item xs={12}>
          <Paper sx={{ p: 3 }}>
            <Typography variant="h6" gutterBottom>
              Playing Statistics
            </Typography>
            <Grid container spacing={2} mt={1}>
              <Grid item xs={6} sm={3}>
                <Box textAlign="center">
                  <Typography variant="h5" color="primary" fontWeight="bold">
                    {stats.vpip}%
                  </Typography>
                  <Typography variant="body2" color="textSecondary">
                    VPIP
                  </Typography>
                </Box>
              </Grid>
              <Grid item xs={6} sm={3}>
                <Box textAlign="center">
                  <Typography variant="h5" color="secondary" fontWeight="bold">
                    {stats.pfr}%
                  </Typography>
                  <Typography variant="body2" color="textSecondary">
                    PFR
                  </Typography>
                </Box>
              </Grid>
              <Grid item xs={6} sm={3}>
                <Box textAlign="center">
                  <Typography variant="h5" color="info.main" fontWeight="bold">
                    {stats.aggression}
                  </Typography>
                  <Typography variant="body2" color="textSecondary">
                    Aggression Factor
                  </Typography>
                </Box>
              </Grid>
              <Grid item xs={6} sm={3}>
                <Box textAlign="center">
                  <Typography variant="h5" color="success.main" fontWeight="bold">
                    {stats.handsPlayed}
                  </Typography>
                  <Typography variant="body2" color="textSecondary">
                    Total Hands
                  </Typography>
                </Box>
              </Grid>
            </Grid>
          </Paper>
        </Grid>
      </Grid>
    </Box>
  );
};
