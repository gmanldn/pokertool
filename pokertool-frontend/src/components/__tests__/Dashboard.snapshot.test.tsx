/**
 * Dashboard Component Snapshot Tests
 * ====================================
 *
 * Snapshot tests detect unintended changes to component rendering.
 * Run `npm test -- -u` to update snapshots when changes are intentional.
 */

import React from 'react';
import { render } from '@testing-library/react';
import { Dashboard } from '../Dashboard';

describe('Dashboard Snapshot Tests', () => {
  it('renders correctly with no messages', () => {
    const { container } = render(<Dashboard messages={[]} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with sample messages', () => {
    const sampleMessages = [
      { id: 1, type: 'info', text: 'Welcome to PokerTool', timestamp: new Date() },
      { id: 2, type: 'success', text: 'Connected to server', timestamp: new Date() },
    ];

    const { container } = render(<Dashboard messages={sampleMessages} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly in loading state', () => {
    const { container } = render(<Dashboard messages={[]} isLoading={true} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
