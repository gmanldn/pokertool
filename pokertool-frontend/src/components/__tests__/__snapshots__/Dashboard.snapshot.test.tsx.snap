// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Snapshot Tests renders correctly in loading state 1`] = `
<div
  class="MuiBox-root css-zxdg2z"
>
  <div
    class="MuiBox-root css-v08z5u"
  >
    <div
      class="MuiBox-root css-j0ozid"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-1313egv-MuiTypography-root"
      >
        Dashboard
      </h4>
      <div
        class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-9td8ez-MuiChip-root"
      >
        <span
          class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
        >
          v100.3.0
        </span>
      </div>
    </div>
    <div
      class="MuiBox-root css-j0ozid"
    >
      <div
        class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorDefault MuiChip-outlinedDefault css-1gyxgoa-MuiChip-root"
      >
        <span
          class="MuiChip-label MuiChip-labelSmall css-rrn746-MuiChip-label"
        >
          Last Update: 3:42:59 AM
        </span>
      </div>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="RefreshIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-1s32w64-MuiPaper-root"
  >
    <div
      class="MuiBox-root css-dppiuw"
    >
      <div
        class="MuiBox-root css-i3pbo"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1lnmg5u-MuiSvgIcon-root"
          data-testid="TableChartIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 10.02h5V21h-5zM17 21h3c1.1 0 2-.9 2-2v-9h-5zm3-18H5c-1.1 0-2 .9-2 2v3h19V5c0-1.1-.9-2-2-2M3 19c0 1.1.9 2 2 2h3V10H3z"
          />
        </svg>
      </div>
      <h6
        class="MuiTypography-root MuiTypography-h6 css-10neesi-MuiTypography-root"
      >
        No Hands Yet
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body2 css-adbk1r-MuiTypography-root"
      >
        Start playing poker to see your hands appear here. The system will automatically detect and record your gameplay.
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1p9xwda-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PlayArrowIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8 5v14l11-7z"
            />
          </svg>
        </span>
        View Dashboard
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Dashboard Snapshot Tests renders correctly with no messages 1`] = `
<div
  class="MuiBox-root css-zxdg2z"
>
  <div
    class="MuiBox-root css-v08z5u"
  >
    <div
      class="MuiBox-root css-j0ozid"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-1313egv-MuiTypography-root"
      >
        Dashboard
      </h4>
      <div
        class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-9td8ez-MuiChip-root"
      >
        <span
          class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
        >
          v100.3.0
        </span>
      </div>
    </div>
    <div
      class="MuiBox-root css-j0ozid"
    >
      <div
        class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorDefault MuiChip-outlinedDefault css-1gyxgoa-MuiChip-root"
      >
        <span
          class="MuiChip-label MuiChip-labelSmall css-rrn746-MuiChip-label"
        >
          Last Update: 3:42:59 AM
        </span>
      </div>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="RefreshIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-1s32w64-MuiPaper-root"
  >
    <div
      class="MuiBox-root css-dppiuw"
    >
      <div
        class="MuiBox-root css-i3pbo"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1lnmg5u-MuiSvgIcon-root"
          data-testid="TableChartIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 10.02h5V21h-5zM17 21h3c1.1 0 2-.9 2-2v-9h-5zm3-18H5c-1.1 0-2 .9-2 2v3h19V5c0-1.1-.9-2-2-2M3 19c0 1.1.9 2 2 2h3V10H3z"
          />
        </svg>
      </div>
      <h6
        class="MuiTypography-root MuiTypography-h6 css-10neesi-MuiTypography-root"
      >
        No Hands Yet
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body2 css-adbk1r-MuiTypography-root"
      >
        Start playing poker to see your hands appear here. The system will automatically detect and record your gameplay.
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1p9xwda-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PlayArrowIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8 5v14l11-7z"
            />
          </svg>
        </span>
        View Dashboard
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Dashboard Snapshot Tests renders correctly with sample messages 1`] = `
<div
  class="MuiBox-root css-zxdg2z"
>
  <div
    class="MuiBox-root css-v08z5u"
  >
    <div
      class="MuiBox-root css-j0ozid"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-1313egv-MuiTypography-root"
      >
        Dashboard
      </h4>
      <div
        class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-9td8ez-MuiChip-root"
      >
        <span
          class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
        >
          v100.3.0
        </span>
      </div>
    </div>
    <div
      class="MuiBox-root css-j0ozid"
    >
      <div
        class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorDefault MuiChip-outlinedDefault css-1gyxgoa-MuiChip-root"
      >
        <span
          class="MuiChip-label MuiChip-labelSmall css-rrn746-MuiChip-label"
        >
          Last Update: 3:42:59 AM
        </span>
      </div>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="RefreshIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-1s32w64-MuiPaper-root"
  >
    <div
      class="MuiBox-root css-dppiuw"
    >
      <div
        class="MuiBox-root css-i3pbo"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1lnmg5u-MuiSvgIcon-root"
          data-testid="TableChartIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 10.02h5V21h-5zM17 21h3c1.1 0 2-.9 2-2v-9h-5zm3-18H5c-1.1 0-2 .9-2 2v3h19V5c0-1.1-.9-2-2-2M3 19c0 1.1.9 2 2 2h3V10H3z"
          />
        </svg>
      </div>
      <h6
        class="MuiTypography-root MuiTypography-h6 css-10neesi-MuiTypography-root"
      >
        No Hands Yet
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body2 css-adbk1r-MuiTypography-root"
      >
        Start playing poker to see your hands appear here. The system will automatically detect and record your gameplay.
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1p9xwda-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PlayArrowIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8 5v14l11-7z"
            />
          </svg>
        </span>
        View Dashboard
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
